subsystems/intakesubsystem.py

```diff
@@ -0,0 +1,140 @@
+from enum import Enum, auto
+from commands2 import Subsystem
+from wpilib import RobotBase
+from wpimath.geometry import Rotation2d
+from ntcore import NetworkTableInstance
+from util.simtalon import Talon
+from util.simcoder import CTREEncoder
+from util.angleoptimize import intakeAccountForSillyEncoder
+
+import constants
+
+
+class IntakeSubsystem(Subsystem):
+    class IntakeState(Enum):
+        Intaking = auto()
+        Score = auto()
+        Scoring = auto()
+        Knock = auto()
+
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)  # basic subsystem boilerplate
+
+        self.pivotEncoder = CTREEncoder(
+            constants.kPivotEncoderID, constants.kPivotEncoderOffset
+        )
+
+        self.intakeMotor = Talon(
+            constants.kIntakeCANID,
+            constants.kIntakeName,
+            constants.kIntakePGain,
+            constants.kIntakeIGain,
+            constants.kIntakeDGain,
+            constants.kIntakeInverted,
+        )
+        self.intakeMotor.setCurrentLimit(constants.kIntakeCurrentLimit)
+        self.pivotMotor = Talon(
+            constants.kPivotCANID,
+            constants.kPivotName,
+            constants.kPivotPGain,
+            constants.kPivotIGain,
+            constants.kPivotDGain,
+            constants.kPivotInverted,
+            moMagicAccel=constants.kPivotAccel,
+            moMagicVel=constants.kPivotVel,
+        )
+        self.pivotMotor.setNeutralMode(Talon.NeutralMode.Brake)
+        # self.positionDebouncer = Debouncer(0.1, Debouncer.DebounceType.kRising)
+
+        self.resetPivot()
+
+        self.state = self.IntakeState.Score
+        self.targetAngle = Rotation2d()
+
+        self.intakeAtPositionPublisher = (
+            NetworkTableInstance.getDefault()
+            .getBooleanTopic(constants.kIntakeAtPositionKey)
+            .publish()
+        )
+        self.pivotAnglePublisher = (
+            NetworkTableInstance.getDefault()
+            .getFloatTopic(constants.kPivotAngleKey)
+            .publish()
+        )
+        self.intakeStatePublisher = (
+            NetworkTableInstance.getDefault()
+            .getStringTopic(constants.kIntakeStateKey)
+            .publish()
+        )
+
+    def periodic(self) -> None:
+        # a lot simpler when there isn't a convoluted intake sequence
+        if self.state == self.IntakeState.Intaking:
```

Comment:
we are in python 3.12, you are doing the equivelent of a switch case, which has been in python since 3.10, and we have nuked support for anything previous to 3.10

revise with a match statement instead for effeciency